### PR TITLE
feat: Add Winget package option

### PR DESCRIPTION
**Pull request by SinnoSong**
Add Winget package option: use `build.ps1` upload `gptcommit.exe` to Winget. 
Use this command:
```powershell
build.ps1 -Token <GitHubPersonalAccessToken> -Version <PublishVersion>
```
Or use Github action like :https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/.github/workflows/winget.yml

### CODE REVIEW

File: package/winget/zurawiki.gptcommit.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+PackageIdentifier: zurawiki.gptcommit
+PackageVersion: <VERSION>
+ReleaseDate: <DATE>
+InstallerType: zip
+Installers:
+  - Architecture: x64
+    InstallerUrl: https://github.com/zurawiki/gptcommit/releases/download/v0.5.6/gptcommit-x86_64-pc-windows-msvc.zip
```
Comment (zurawiki) [inline package/winget/zurawiki.gptcommit.installer.yaml:7]: We should avoid hard-coding the version numbers here. Can we use the `<VERSION>` pattern here and below?
```suggestion
    InstallerUrl: https://github.com/zurawiki/gptcommit/releases/download/v<VERSION>gptcommit-x86_64-pc-windows-msvc.zip
```

File: package/winget/zurawiki.gptcommit.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    InstallerSha256: <HASH>
+    NestedInstallerType: portable
+    NestedInstallerFiles:
+      - RelativeFilePath: .\gptcommit.exe
+        PortableCommandAlias: gptcommit
+  - Architecture: x86
+    InstallerUrl: https://github.com/zurawiki/gptcommit/releases/download/v0.5.6/gptcommit-x86_64-pc-windows-msvc.zip
```
Comment (zurawiki) [inline package/winget/zurawiki.gptcommit.installer.yaml:14]: Same here

**Review by zurawiki — CHANGES_REQUESTED** [review]

File: package/winget/build.ps1 [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+# Download Winget-Create msixbundle, install, and execute update.
+$appxBundleFile = "$env:TEMP\wingetcreate.msixbundle"
+Invoke-WebRequest https://aka.ms/wingetcreate/latest/msixbundle -OutFile $appxBundleFile
+Add-AppxPackage $appxBundleFile
+
+# Create PR
+# wingetcreate submit --token <GitHubPersonalAccessToken> <PathToManifest>
+wingetcreate submit --token $Token $Version
```
Comment (zurawiki) [inline package/winget/build.ps1:63]: What permissions does the Github token need here?

File: package/winget/zurawiki.gptcommit.installer.yaml (hunk 1 above)
Comment (SinnoSong) [reply package/winget/zurawiki.gptcommit.installer.yaml]: Yes, it can be replaced.

File: package/winget/build.ps1 (hunk 3 above)
Comment (SinnoSong) [reply package/winget/build.ps1:63]: This Github token is use for creating a PR in https://github.com/microsoft/winget-pkgs

**Comment by SinnoSong** [comment]
@zurawiki I replaced hard-coding version

**Comment by zurawiki** [comment]
The rest of the PR looks ok. How are we running this github action then? Should I copy workflow/winget.yml into this repo?

**Comment by SinnoSong** [comment]
@zurawiki Yes, you need copy workflow/winget.yml into this repo. I think some codes should be modified:
- The `working-directory` in line 15 should change to `${{ github.workspace }}/package/winget`.
- The `WINGETCREATE_TOKEN` in line 17, you should see this document: https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#triggering-a-workflow-from-a-workflow . You need to [creating a personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) and [encrypted secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets)

**Comment by slacksystem** [comment]
Sorry if this is obvious but I did `winget search gptcommit` and got zero results. Am I missing something? I thought the latest release said it was implemented but I may be going about this wrong

**Comment by SinnoSong** [comment]
@slacksystem There is no one PR be merged: https://github.com/microsoft/winget-pkgs/pulls?q=gptcommit

**Pull request merged**